File: src/main/java/bruteforce/boj_22864/Main.java
package bruteforce.boj_22864;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

// 14184kb 104ms
public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int tired = Integer.parseInt(st.nextToken());
        int work = Integer.parseInt(st.nextToken());
        int release = Integer.parseInt(st.nextToken());
        int maxTired = Integer.parseInt(st.nextToken());
        int tiredSum = 0;
        int result = 0;
        for (int i=0;i<24;i++) {
            if (tiredSum + tired <= maxTired) {
                result += work;
                tiredSum += tired;
            } else {
                tiredSum = Math.max(0, tiredSum - release);
            }
        }
        System.out.print(result);
    }
}

File: src/main/java/BruteForce/boj_22864/Main.kt
package bruteforce.boj_22864

import java.io.BufferedReader
import java.io.InputStreamReader
import kotlin.math.max

// 20752kb 108ms
fun main(args: Array<String>) {
	val br = BufferedReader(InputStreamReader(System.`in`))
	val (damage,work,heal,deadline) = br.readLine().split(" ").map {it.toInt()}
	var damageSum = 0
	var result = 0
	for (i in 0..23) {
		if (damage + damageSum <= deadline) {
			damageSum += damage
			result += work
		} else {
			damageSum = max(0, damageSum-heal)
		}
	}
	println(result)
}